### PR TITLE
CLI: Support home workspace through `~`

### PR DESCRIPTION
**Pull request by davidfestal**
## Summary

Introduce a way to go to the user Home workspace doing:
```
kubectl workspace '~'
```
This PR depends on PR #1373 which implements the feature on the KCP server side.

## Related issue(s)

Part of the [User Home Workspaces EPIC](#1069)

### CODE REVIEW

File: pkg/cliplugins/workspace/plugin/kubeconfig.go [hunk 1]
```diff
@@ -181,6 +181,13 @@ func (kc *KubeConfig) UseWorkspace(ctx context.Context, name string) error {
 		u.Path = path.Join(u.Path, parentClusterName.Path())
 		newServerHost = u.String()
 
+	case "~":
+		homeWorkspace, err := kc.clusterClient.Cluster(tenancyv1alpha1.RootCluster).TenancyV1beta1().Workspaces().Get(ctx, "~", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		newServerHost = homeWorkspace.Status.URL
+
```
Comment (sttts) [inline pkg/cliplugins/workspace/plugin/kubeconfig.go:190]: no unit test?

Comment (davidfestal) [reply pkg/cliplugins/workspace/plugin/kubeconfig.go:190]: Add a test case in commit a31a51911357a1c75be62d8fa6817d61f047b860

**Comment by davidfestal** [comment]
@sttts is the added test OK for you ?

**Comment by sttts** [comment]
Totally unrelated errors. Merging.

**Pull request merged**